scripts/core/editor3/helpers/handleBeforeInputHighlights.ts

```diff
@@ -28,7 +36,7 @@ function getRelevantCharactersForCollapsedSelection(editorState, collapsedSelect
 
     characters = characters.concat(getDraftCharacterListForSelection(editorState, resizedSelection));
 
-    if (resizeRightFailed === true) {
+    if (resizeRightFailed === true || selectionAtTheEndOfTheBlock) {
```

Comment:
Could you briefly explain what was the main problem? Why would it expand the annotation? Could you also rename `getRelevantCharactersForCollapsedSelection` so it's clear what does it do? (if you understand it :))

Reply:
Sure, basically `getRelevantCharactersForCollapsedSelection` returns the characters before and after the cursor so we can check the inline styles of those.

If it fails resizing the selection to the right, it will return a character with an empty style (which prevents extending inline styles while writing at the end). The problem was that resizing to the right was not failing if there was another block after it, even though the cursor was at the end of the line. So I added that check to extend that behaviour.

I'm thinking now that I might have to check if the user is pressing space, because if they're not they might want to extend the highlight. I'll check with QA.

Reply:
Ok, the "right" behaviour should be the one we already have with one line, so this PR should be ready

Reply:
>  returns the characters before and after the cursor so we can check the inline styles of those.

> If it fails resizing the selection to the right, it will return a character with an empty style (which prevents extending inline styles while writing at the end).

Can you add this to the code? :)

> The problem was that resizing to the right was not failing if there was another block after it

How about resizing the selection by 2 chars to the right if it's at the end of the block. It seems like the same thing was done with resizeLeft, it would be good to keep the implementations the same if they do the same thing.

> `const resizeLeft = selectionAtTheStartOfTheLine ? 2 : 1;` // SDESK-2861

> I'm thinking now that I might have to check if the user is pressing space, because if they're not they might want to extend the highlight.

What if they want to extend it by adding another word or want to add punctuation which shoulnd't extend the annotation? I think it's best not to allow it and keep it consistent, that if you want to add anything to the annotation, you move one char before the end, write what you want and remove the last char.

If you were to check for space

Reply:
> behaviour should be the one we already have with one line

What behavior are you referring to?

Reply:
The one we have on master already when we only have one block (the one you show in [this video](https://drive.google.com/file/d/1S0vKay-l9HWagqrEunuSNV4OKpTL6TS4/view))